zerver/views/users.py

```diff
@@ -491,7 +491,7 @@ def add_bot_backend(
                 "Please contact your server administrator."
             )
         )
-    except ValueError:
+    except ValueError:  # nocoverage
         raise JsonableError(_("Bad name or username"))
```

Comment:
I am not sure how can we reproduce this but would be better if we can find how this can be reproduced and add tests for this. It was probably because of using `"my\nbot"` as short name from what I can understand from the tests you modified above. I am not sure about how we should move forward though, so we can wait for feedback from Tim and others.

Reply:
Okay, is there anything I need to re-work?

Reply:
I don't understand why we're adding that regular expression at all; the regular expression is very hard to read, AFAIK may have bugs in it since I don't understand it, whereas here we're using the language's standard email address validator. We should only do one of the two checks, since they're redundant, but the existing `Address()` check is the correct one to keep.

Reply:
Hmm, I agree regular expression is difficult to understand and should be avoided.

But yeah we should at least have consistent checks in both frontend and backend I think. As per the current checks, "@" is not allowed in email field when creating bots using webapp UI but if I remove the check (i.e. the `is_local_part` code), I can create bots having "@" in email.